core/src/main/java/org/stellar/anchor/sep24/Sep24Service.java

```diff
@@ -420,10 +419,13 @@ public InteractiveTransactionResponse deposit(Sep10Jwt token, Map<String, String
     }
 
     String quoteId = depositRequest.get("quote_id");
+    AssetInfo sellAsset = assetService.getAssetByName(depositRequest.get("source_asset"));
     if (quoteId != null) {
-      AssetInfo sellAsset = assetService.getAssetByName(depositRequest.get("source_asset"));
-      this.validatedAndPopulateQuote(
+      validateAndPopulateQuote(
           quoteId, sellAsset, asset, strAmount, builder, DEPOSIT.toString(), txnId);
+    } else {
+      builder.amountExpected(strAmount);
+      if (sellAsset != null) builder.amountInAsset(sellAsset.getSep38AssetName());
```

Comment:
Looks like a bug -> we not set amountInAsset anymore for when quote is provided. Should this be lifted outside of the parent "if" statement? (I.e. move it to the line below 422)

Reply:
The `validateAndPopulateQuote` function sets `amountInAsset`.

Reply:
Ah you are right I saw deleted code and for some reason I thought it didn't do that